frontend/apps/filemanager/filemanager.lua

```diff
@@ -157,6 +157,30 @@ function FileManager:init()
                         UIManager:close(self.file_dialog)
                     end,
                 },
+                {
+                    text = _("Purge .sdr"),
+                    enabled = lfs.attributes(util.realpath(file), "mode") == "file" and util.realpath(file):match("%.") and
+                            lfs.attributes(util.realpath(file):match("(.*)%.")..".sdr", "mode") == "directory",
```

Comment:
> For getSidecar() returning nil, wouldnt we need to check for side effects in all the existing code, they probably expect a string.

The existing code is expecting a string. However, for this case, we changed the behavior from crashing to returning `nil`, which is a little bit better. But your point is valid, we need to double check all the callers of getSidecar. I think `getSidecar` is probably a helper function that should be hidden from most of the application code because it really does not have any correctness guarantee, it's simply a short cut for string manipulation. So we should check if we can replace its usage with `hasSidecarDir` as much as possible.

> Or just define how should be named the sidecar dir for a file with no extension we will never meet (just add .sdr ?)

Hmm.. I don't see a usecase for having a .sdr for files without extension right now. If you can think of any, then we should totally do it.